File: backend/src/main/java/net/zorphy/backend/main/game/dto/stats/correlation/CorrelationResult.java
package net.zorphy.backend.main.game.dto.stats.correlation;

import java.util.List;

public record CorrelationResult(
        CorrelationMetadata metadata,
        double coefficient,
        double slope,
        double intercept,
        List<CorrelationDataPoint> points
) {
}

File: frontend/src/app/main/games/dto/stats/correlations/CorrelationResult.ts
import {CorrelationDataPoint} from "./CorrelationDataPoint";
import {CorrelationMetadata} from "./CorrelationMetadata";

export interface CorrelationResult {
    metadata: CorrelationMetadata,
    coefficient: number,
    slope: number,
    intercept: number,
    points: CorrelationDataPoint[]
}